.github/workflows/codeql.yml

```diff
@@ -1,74 +0,0 @@
-# For most projects, this workflow file will not need changing; you simply need
```

Comment:
Deleting this CodeQL workflow because I couldn't figure out how to make it run on Java 17. Its Java version autodetection was failing on our parent POM and I couldn't seem to override it. Perhaps this can be restored once our parent POM is cleaned up and requires 2.463 or newer.

Reply:
OK, feel free to disable it temporarily on the main branch and I can merge those changes into this PR.

Reply:
there is no changes in git to disable the workflow. The change is made from the GitHub UI, it is already done.

Reply:
Thanks @kuisathaverat! I've reverted this deletion and the CI build is now passing. I think this is now ready to merge and release.